### ISSUE TITLE
[Suggestion] Configurable time boxes

### ISSUE THREAD

**Issue by tlahn** [vukani-dev/improvement-roll#18]
Hi there,

I really like the software and the possibility to roll by available time is one of the core features I am happy about.

For me personally, I could use a time box 0-5 minutes because I regularly do 5 minutes breaks and one box of about 1h - 4h when there is a free afternoon or long evening, while 4h+ only makes sense when I have at least a full day available.

I guess these timeboxes might be very individual for each user and providing too many of them per default could overwhelm new users. Therefore I thought whether it would be possible to make these configurable. For example instead of storing time categories per task, it could be a time effort in minutes.

This could open the possibility to roll a task based on any number of available minutes. But also this adds some complexity in task selection for the rolling algorithm. Like: I have 45 minutes available, so it would be good to have any task that is 0-45 minutes, but it might be desirable to have one that rather fills most of the time instead of re-rolling every 5 minutes. On the other hand, it might also be a valuable time for doing a series of shorter tasks that are not completed often. But this may as well be another extension.

What do you think about this?

**Comment by vukani-dev** [comment]
Hello! Glad to hear you like the app! 

So I think I understand what you're saying, and its a great point.
There may be users who want to differentiate 5 minute tasks from 10 minute tasks for example.
It would be nice to have a system that's custom for each persons use-case, and I think the method you described is a great way to do that (giving tasks an effort point value in minutes rather than a redefined range). 

All that's left is sorting out the rolling algorithm to accommodate this. As you noted, if I say I have 45 minutes available I could either want something that will closely fill up that time, or I would be fine with the "luck of the draw" aspect of maybe getting a short 5 minute task.

Maybe people who are fine with the "luck of the draw" aspect are better off disabling the time sensitivity in a new category specific for that? That way we can strictly leave the time sensitive categories to be run by giving you a task close to your available time (within a range of course. If you only have one 45 minute task you might want a 40 minute task mixed in the randomizer so you don't get the same one all the time.)

Long story short, I thank you for bringing this up. On reflection I think its pretty important. And I would want something like this before releasing the Community Categories feature I've been working on. As it would mean re-hauling how tasks are saved and work.

Let me know if I've understood you properly!

**Comment by tlahn** [comment]
Hi,
yes, this is exactly what I meant.

For example (snippet from default categories):

```
...
{
      name: "Meditate for 5 minutes",
      desc:
        "Find a quiet spot and set a timer for 5 minutes. Close your eyes and focus on your breath, try not to think about anything. Do this until the timer runs out",
      timeInMinutes: 5,
    },
    {
      name: "Meditate for 10 minutes",
      desc:
        "Find a quiet spot and set a timer for 10 minutes. Close your eyes and focus on your breath, try not to think about anything. Do this until the timer runs out",
      timeInMinutes: 10,
    },
...
```

**Comment by vukani-dev** [comment]
This feature has been added in the newest update. Please check out the README on more details and let me know if any problems arise!
Thank you!!
